subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -45,6 +45,15 @@ static u32_t scan_result;
 
 static struct net_mgmt_event_callback wifi_shell_mgmt_cb;
 
+#define print(shell, level, fmt, ...)					\
```

Comment:
So, I was thinking of the same type of solution; but then thought, if printk() is equivalent to shell_fprintf(), then why not just stay with printk()?   And if they're not equivalent, then maybe there needs to be some kind of init() function for shells to initialize the context.shell only once for each shell module, so that shell_fprintf() can be used consistently.

Reply:
The printk() and shell_fprintf() both print something to somewhere. With shell_fprintf() it is possible to select the backend where the output will go, printk() will just print something to console. Note that the console might not be the same as the shell where user is connected.

At some point there will be a "default" shell and when that happens the printk() should be changed to use that.